File: static/command/js/ext-js/app/model/DataSource.js
Ext.define('command.model.DataSource', {
    extend: 'Ext.data.Model',
    idProperty: 'id',
    fields: [
        {name: 'id',  type: 'string'},
        {name: 'source_name',   type: 'string'},
        {name: 'python_class',  type: 'string'},
        {name: 'is_local',  type: 'boolean'}
    ]
});

File: command/lib/db/compendium/data_source.py
from django.db import models


class DataSource(models.Model):
    source_name = models.TextField()
    python_class = models.TextField()
    is_local = models.BooleanField(default=False)

    def to_dict(self):
        fields = ['id', 'source_name', 'python_class', 'is_local']
        return {k: self.__dict__[k] for k in fields}





